### PR TITLE
Output panel optimisation

### PR DESCRIPTION
**Pull request by davegarthsimpson**
### Motivation
Output panel prints slower than the CLI and on windows is out of sync with the upload operation.

### Change description
Adds a simple buffer mechanism for the related stream from the CLI.

### Other information
Optimisations on CLI side should also complement the resolution of this issue.

### Reviewer checklist

* [x] PR addresses a single concern.
* [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [x] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: arduino-ide-extension/src/node/utils/simple-buffer.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export class SimpleBuffer {
+  chunks: Uint8Array[] = [];
```
Comment (kittaakos) [inline arduino-ide-extension/src/node/utils/simple-buffer.ts:2]: Missing `private`?

File: arduino-ide-extension/src/node/utils/simple-buffer.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        onFlush(chunkString);
+      }
+    }, flushTimeout);
+  }
+
+  public addChunk(chunk: Uint8Array): void {
+    this.chunks.push(chunk);
+  }
+
+  private clearChunks() {
```
Comment (kittaakos) [inline arduino-ide-extension/src/node/utils/simple-buffer.ts:21]: Missing return type?

File: arduino-ide-extension/src/node/core-service-impl.ts [hunk 3]
```diff
@@ -39,6 +40,8 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
 
   protected uploading = false;
 
+  private FLUSH_OUTPUT_MESSAGES_TIMEOUT_MS = 32;
```
Comment (kittaakos) [inline arduino-ide-extension/src/node/core-service-impl.ts:43]: Missing readonly? If this is a static constant, it should be `static`. Or please move out from the class to the module scope and use `const`. Having a variable in the class with `SUCH_NAMING` is confusing. Thank you!

**Review by per1234 — APPROVED** [review]
I tested with the build from https://github.com/arduino/arduino-ide/pull/1058/commits/085f97e9989ea891c0581b822c261c3fa02179a7 and this fixes both https://github.com/arduino/arduino-ide/issues/68 and https://github.com/arduino/arduino-ide/issues/405 for me.

In order to do a more thorough test of https://github.com/arduino/arduino-ide/issues/68, I replaced the standard upload tool command with a script that prints 10000 lines of text. That results in a large delay between the completion of the output process and the completion of printing to the Output panel when using the latest build from the `main` branch. Using the build from this PR, the output printing completes at the same time as the output process.

Thanks Dave!

**Pull request merged**